internal/controllers/utils/utils.go

```diff
@@ -32,6 +33,12 @@ func MatchesIngressClass(obj client.Object, controllerIngressClass string, isDef
 			return true
 		}
 	}
+	// For KongCustomEntities, we check whether the `spec.ControllerName` matches.
+	if customEntity, isKongCustomEntity := obj.(*kongv1alpha1.KongCustomEntity); isKongCustomEntity {
+		if customEntity.Spec.ControllerName == controllerIngressClass {
+			return true
+		}
+	}
```

Comment:
This should also be guarded with the feature gate

Reply:
I don't think this should be put behind the feature gate because only the KCE controller will call this and when the feature gate is disabled, the controller is not enabled. Also, there is no place to pass a feature gate (or an extra parameter) here, and chaning the sign of the function may have too much affects on other controllers.